applications/GeoMechanicsApplication/custom_workflows/time_step_executor.cpp

```diff
@@ -21,31 +21,31 @@ void TimeStepExecutor::SetSolverStrategy(std::shared_ptr<StrategyWrapper> Solver
     mStrategyWrapper = std::move(SolverStrategy);
 }
 
-void TimeStepExecutor::SetProcessReferences(ProcessRefVec ProcessRefs)
+void TimeStepExecutor::SetProcessObservables(const std::vector<std::weak_ptr<Process>>& rProcessObservables)
 {
-    mProcessRefs = std::move(ProcessRefs);
+    mProcessObservables = rProcessObservables;
 }
 
 TimeStepEndState TimeStepExecutor::Run(double Time)
 {
     mStrategyWrapper->Initialize();
     mStrategyWrapper->InitializeSolutionStep();
 
-    for (const auto& process : mProcessRefs)
+    for (const auto& process_observable : mProcessObservables)
     {
-        process.get().ExecuteInitializeSolutionStep();
+        auto process = process_observable.lock();
+        process->ExecuteInitializeSolutionStep();
     }
 
     mStrategyWrapper->Predict();
     mStrategyWrapper->SolveSolutionStep();
 
-    for (const auto& process : mProcessRefs)
+    for (const auto& process_observable : mProcessObservables)
     {
-         process.get().ExecuteFinalizeSolutionStep();
+        auto process = process_observable.lock();
+        process->ExecuteFinalizeSolutionStep();
```

Comment:
I think we need to check whether `process` is not empty. Note that `lock` returns a `shared_ptr` object that temporarily partakes in ownership of the managed object (see [this page](https://en.cppreference.com/w/cpp/memory/weak_ptr/lock)). Now when the managed object has been destructed, we get an empty `shared_ptr`, resulting in undefined behavior when attempting to access it.
```suggestion
        if (process) {
            process->ExecuteFinalizeSolutionStep();
        }
```
We may consider to issue a warning when the managed object no longer exists, since that was probably not intended.

This comment also applies to the other loop over the process observables.

Reply:
Done.